File: hello-java/audio-stream-client/src/main/java/org/feuyeux/mmap/audio/client/util/ErrorHandler.java
package org.feuyeux.mmap.audio.client.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Centralized error handling for the audio stream client.
 * Provides consistent error handling, logging, and recovery strategies.
 * Matches the C++ ErrorHandler interface.
 */
public class ErrorHandler {
    private static final Logger logger = LoggerFactory.getLogger(ErrorHandler.class);

    /**
     * Error types enumeration.
     */
    public enum ErrorType {
        CONNECTION_ERROR,
        FILE_IO_ERROR,
        PROTOCOL_ERROR,
        TIMEOUT_ERROR,
        VALIDATION_ERROR
    }

    /**
     * Report an error.
     *
     * @param type        error type
     * @param message     error message
     * @param context     additional context
     * @param recoverable whether the error is recoverable
     */
    public void reportError(ErrorType type, String message, String context, boolean recoverable) {
        String logMessage = String.format("[%s] %s - Context: %s", type.name(), message, context);

        if (recoverable) {
            logger.warn(logMessage);
        } else {
            logger.error(logMessage);
        }
    }

    /**
     * Report an error without context.
     *
     * @param type    error type
     * @param message error message
     */
    public void reportError(ErrorType type, String message) {
        reportError(type, message, "", false);
    }
}

File: hello-kotlin/src/main/kotlin/client/util/ErrorHandler.kt
package client.util

import Logger

/**
 * Error handling utilities
 */
object ErrorHandler {
    /**
     * Handle an error with logging and optional recovery
     */
    fun handleError(message: String, exception: Exception? = null): Nothing {
        Logger.error(message)
        if (exception != null) {
            Logger.error("Exception: ${exception.message}")
            exception.printStackTrace()
        }
        throw exception ?: Exception(message)
    }
    
    /**
     * Log a warning without throwing
     */
    fun logWarning(message: String) {
        Logger.warn(message)
    }
    
    /**
     * Wrap a block with error handling
     */
    inline fun <T> withErrorHandling(operation: String, block: () -> T): T {
        return try {
            block()
        } catch (e: Exception) {
            handleError("Error during $operation", e)
        }
    }
}
